File: COMP_HW1/COMP_HW1_cpp/Person.cpp
#include "Person.h"
#include <string>
#include <sstream>
using namespace std;

Person::Person(string a, string b, int c){
  firstname = a;
  lastname = b;
  phonenumber = c;
}

void Person::setFirstName(string first){
  firstname = first;
}

string Person::getFirstName(){
  return firstname;
}

void Person::setLastName(string last){
  lastname = last;
}

string Person::getLastName(){
  return lastname;
}

void Person::setPhoneNumber(int number){
  phonenumber = number;
}

int Person::getPhoneNumber(){
  return phonenumber;
}

void Person::print(){
  cout << firstname << " " << lastname << "_";
  string call;

  //convert integer value to string
  stringstream ss;
  ss << phonenumber;
  call = ss.str();

  if (phonenumber >= 1000000000) //010-XXXX-XXXX
	call = "0" + call.substr(0,2) + "-" + call.substr(2,4)
	       + "-" + call.substr(6, 4);
  else //02-XXXX-XXXX
	call = "0" + call.substr(0,1) + "-" + call.substr(1,4)
	       + "-" + call.substr(5, 4);

  cout << call;
}

File: COMP_HW1/COMP_HW1_java/Person.java
public class Person {
	
	private String firstName;
	private String lastName;
	private int phoneNumber;
	
	Person(String a, String b, int c){
		firstName = a;
		lastName = b;
		phoneNumber = c;
	}
	
	public void setFirstName(String a){
		firstName = a;
	}
	
	public String getFirstName(){
		return firstName;
	}
	
	public void setLastName(String a){
		lastName = a;
	}
	
	public String getLastName(){
		return lastName;
	}
	
	public void setPhoneNumber(int i){
		phoneNumber = i;
	}
	
	public int getPhoneNumber(){
		return phoneNumber;
	}
	
	void print(){
		System.out.print(firstName + " " + lastName + "_");
		String call = Integer.toString(phoneNumber); //convert integer value to string
		if(phoneNumber >= 1000000000){ //010-XXXX-XXXX
			call = "0" + call.substring(0, 2) + "-" + 
			       call.substring(2, 6) + "-" + call.substring(6);
		}else{ //02-XXXX-XXXX
			call = "0" + call.substring(0, 1) + "-" + 
			       call.substring(1, 5) + "-" + call.substring(5);
		}
		System.out.print(call);
	}
	
}
